fixes #26198 - ensure host list does not require full page reload

ohadlevy opened the pull request.

ohadlevy committed:
fixes #26198 - ensure host list does not require full page reload

ekohl commented on webpack/assets/javascripts/hosts/tableCheckboxes.js:
```diff
@@ -0,0 +1,249 @@
+import { sprintf, n__, translate as __ } from '../react_app/common/I18n';
+
+// Array contains list of host ids
+$.cookieName = `_ForemanSelected${window.location.pathname.replace(/\//, '')}`;
```
Don't you need to import `$`?

ohadlevy replied:
thanks, updated, and also moved the deprecated jquery.cookie next to the code.

ohadlevy commented:
I've added depreciation functions as it seems discovery / scap uses it.

ezr-ondrej commented on webpack/assets/javascripts/bundle.js:
```diff
@@ -30,6 +29,7 @@ import compute from './foreman_compute_resource';
 import componentRegistry from './react_app/components/componentRegistry';
 import i18n from './react_app/common/I18n';
 import * as foremanDocument from './react_app/common/document';
+import hosts from './foreman_hosts';
```
Why not to import all of the dependencies, but just one? Looks bit weird now.

ohadlevy replied:
this allow to consume the functions both via tfm.hosts.. and as a simple import {x} from ...

I did it in order to not break code base, and its a pattern we already have (e.g. see compute resources).

ezr-ondrej replied:
oh, ok, make sense, thanks for explanation :+1:

ezr-ondrej commented on webpack/assets/javascripts/foreman_hosts.js:
```diff
@@ -1,4 +1,13 @@
 import { uniq, set } from 'lodash';
+import * as table from './hosts/tableCheckboxes';
+
+export default {
```
Not sure if this is preferred way of exporting, maybe it is - it is stating at the beginning all the exports. but I would deleted the other exports than - why to export same stuff twice? :thinking:

ezr-ondrej commented on webpack/assets/javascripts/hosts/tableCheckboxes.js:
```diff
@@ -0,0 +1,251 @@
+
+// Array contains list of host ids
+$.cookieName = `_ForemanSelected${window.location.pathname.replace(/\//, '')}`;
+$.foremanSelectedHosts = readFromCookie();
```
I know, you are just moving it, not writing, but this assign seems especially fishy, maybe would be nice to rewrite it?
Do not understand the purpose, but assigning some property to the jquery doesn't sound like a good idea, I would suggest to put it at least in `window` in this PR.

ezr-ondrej reviewed (commented):
Looks working, just the host selection preservation stopped working (the checkboxes are not staying checked after taking an action).

ohadlevy commented:
I've:
- rebased
- fixed the issue where content was no longer visible (after #6502)
- refactored $. variables
- moved foreman_url function to foremanTools
- used foremanUrl function
- fixed the issue with checkboxes not being persistent

ohadlevy commented on webpack/assets/javascripts/hosts/tableCheckboxes.js:
```diff
@@ -0,0 +1,254 @@
+}
+
+function paginationMetaData() {
+  const pagination = $('#pagination');
```
@glekner @amirfefer this seems to be broken after pagination refactoring, its outside the scope of this PR, so its mostly FYI as I've encountered it.

timogoebel replied:
Do we have a redmine issue to fix this?

ohadlevy replied:
https://projects.theforeman.org/issues/26494

ohadlevy commented:
hmm.. test failures do not seem related.

ohadlevy commented:
[test foreman]

ohadlevy commented:
rebased and removed the workaround for showing the content as it was fixed in #6545

ohadlevy commented:
tests failures do not seem related, @tbrisker mind having a look? thanks.

ohadlevy commented:
anything else needed for this PR? thanks.

ezr-ondrej reviewed (dismissed):
Code looks good to me and what I could come up with to test is working!
Good job @ohadlevy! :+1:

ares commented:
rebase needed now

ohadlevy commented:
rebased.

ohadlevy commented:
[test foreman]
[test katello]

ohadlevy commented:
@tbrisker do you mind reviewing? I believe its all ready for a while now.

amirfefer reviewed (dismissed):
Tested and works well (excluding the choose all host regression :( )
The hosts page is loading faster, and overall it's a great change 
Thanks @ohadlevy  :+1:

ohadlevy commented:
anyone else can merge? perhaps @timogoebel ?

ohadlevy commented:
test failures not related.

timogoebel commented on webpack/assets/javascripts/hosts/tableCheckboxes.js:
```diff
@@ -0,0 +1,252 @@
+
+// triggered by a host checkbox change
+export function hostChecked({ id, checked }) {
+  const multpleAlert = $('#multiple-alert');
```
typo: should be `multipleAlert` (missing `i`)

timogoebel reviewed (changes requested):
@ohadlevy: I found one small typo, otherwise: 👍

ohadlevy commented:
@timogoebel updated - thanks, test failures are unrelated.

timogoebel commented:
Thanks, @ohadlevy.
